src/oemof/solph/components/_link.py

```diff
@@ -74,14 +78,30 @@ class Link(on.Transformer):
     0.8
     """
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
+    def __init__(
+        self,
+        label,
+        inputs,
+        outputs,
+        conversion_factors,
+    ):
+        if inputs is None:
+            warn_if_missing_attribute(self, "inputs")
+            inputs = {}
+        if outputs is None:
+            warn_if_missing_attribute(self, "outputs")
+            outputs = {}
+        if conversion_factors is None:
+            warn_if_missing_attribute(self, "conversion_factors")
+            conversion_factors = {}
```

Comment:
```suggestion
        if inputs is None:
            inputs = {}
        if outputs is None:
            outputs = {}
        if conversion_factors is None:
            conversion_factors = {}
```
As this won't happen accidentally, I would not warn. Whoever sets it to Null on purpose, should be know what they are doing.